### ISSUE TITLE
feature request: skipInitialTransitionIn

### ISSUE THREAD

**Issue by zenflow** [illinois/next-page-transitions#8]
I want page transitions, but only *between* pages.. not on the initial page load

because:
- I want that snappy/fast first meaningful paint of next.js, thanks to the server-rendered markup that is displayed before any js is even run.
- For SEO, I want to be certain that all robots recognize the contents as visible. I know the contents are rendered server-side, but with the styles it isn't visible, and I believe some intelligent crawlers ignore content that is not in a visible state.

I propose supporting this this via a new boolean prop `skipInitialTransitionIn`. 

Without this behavior, the benefits of SSR seem to be defeated.

@illinois what do you think? is it possible?

**Comment by nwalters512** [comment]
Sure, definitely possible. I don't have much free time at present to work on this, but I'd welcome a proposed API and a pull request implementing this!

**Comment by myleshyson** [comment]
Hey at least for the SEO part, I found that if you set loaded in getInitialProps, then you can control whether your content loads from server or from client. For example.

```js
        static pageTransitionDelayEnter = true;
	static defaultProps = {
		loaded: false
	}

	constructor(props) {
		super(props);
		this.state = { loaded: props.loaded, gameModalVisible: false };
	}
	static async getInitialProps() {
		let load;
		if(typeof window === 'undefined') {
			load = true;
		} else {
			load = false;
		}
		return {loaded: load};
	}
	componentDidMount() {
		this.timeoutId = setTimeout(() => {
			this.props.pageTransitionReadyToEnter();
			this.setState({ loaded: true });
		}, 300);
	}

	componentWillUnmount() {
		if (this.timeoutId) clearTimeout(this.timeoutId);
	}
	render() {
		if (!this.state.loaded) return null;
                return <div>Content!</div>
        }

```

When I set it up this way, when I crawled the site through Screaming Frog it picked up all of my content, meta tags, etc.

**Comment by mstefaniakap** [comment]
I've created PR https://github.com/illinois/next-page-transitions/pull/29. Hope it'll help and could be reviewed and merged soon.
Usage is simple - if you want disable initial transition simply set `skipInitialTransition` prop to `false`, component could looks like:
```
<PageTransition skipInitialTransition={true} timeout={500} classNames="page-transition">
  <Component {...pageProps} />
</PageTransition>
```

**Comment by nwalters512** [comment]
Implemented in #29
